src/mlpack/core/data/split_data.hpp

```diff
@@ -306,18 +314,20 @@ void Split(const arma::Mat<T>& input,
  * @return std::tuple containing trainData (arma::Mat<T>), testData
  *      (arma::Mat<T>), trainLabel (arma::Row<U>), and testLabel (arma::Row<U>).
  */
-template<typename T, typename U>
-std::tuple<arma::Mat<T>, arma::Mat<T>, arma::Row<U>, arma::Row<U>>
+template<typename T, typename LabelsType,
+         typename = std::enable_if_t<arma::is_arma_type<LabelsType>::value ||
+                          arma::is_Mat_only<LabelsType>::value> >
```

Comment:
Same here---I think `is_Mat_only` is redundant here.  Correct me if I'm wrong. :+1:

Reply:
I will fix this too 👍